service/src/main/java/com/codeforcommunity/processor/ProcessorImpl.java

```diff
@@ -13,66 +16,129 @@
-  // id should really not be a static number! see down below for dynamic id, hash of username
-  //private int id = 10;
+  // id should really not be a static number! see down below for dynamic id, hash
+  // of username
+  // private int id = 10;
 
   public ProcessorImpl(DSLContext db) {
     this.db = db;
   }
 
   @Override
-  public List<MemberReturn> getAllMembers() {
-    List<Member> members = db.selectFrom(Tables.MEMBER).fetchInto(Member.class);
-    return members.stream().map(member -> new MemberReturn(member.getFirstName(), member.getLastName()))
+  public List<UserReturn> getAllUsers() {
+    List<Users> users = db.selectFrom(Tables.USERS).fetchInto(Users.class);
+    return users.stream()
+        .map(user -> new UserReturn(user.getId(), user.getEmail(), user.getFirstName(), user.getLastName(),
+            user.getGraduationYear().toString(), user.getMajor(), user.getPrivilegeLevel()))
         .collect(Collectors.toList());
   }
 
   @Override
-  public boolean attendMeeting(String meetingid, String username) {
-    Result meetingResult = db.fetch("select * from meeting where id=?;", meetingid);
-    // for now assume username is just the first name
-    Result memberResult = db.fetch("select id from member where first_name=?;", username);
+  public List<UserReturn> getEventUsers(int eventId) {
+    //List<Users> users = db.selectFrom(Tables.USERS).fetchInto(Users.class);
+    List<Users> users = db.fetch("SELECT * FROM users CROSS JOIN (SELECT * FROM event_check_ins where id = ?) AS x;", eventId)
+        .into(Users.class);
+    return users.stream()
+        .map(user -> new UserReturn(user.getId(), user.getEmail(), user.getFirstName(), user.getLastName(),
+            user.getGraduationYear().toString(), user.getMajor(), user.getPrivilegeLevel()))
+        .collect(Collectors.toList());
+  }
+  @Override
+  public boolean attendEvent(String eventCode, int userid) {
+    try{
+    Result eventResult = db.fetch("select * from events where code = ?;", eventCode);
 
     DateTimeFormatter formatter = DateTimeFormatter.ofPattern("yyyy-MM-dd HH:mm:ss.S");
-    LocalDateTime date = LocalDateTime.parse(meetingResult.getValue(0, "date").toString(), formatter);
-    boolean open = Boolean.parseBoolean(meetingResult.getValue(0, "open").toString());
+    LocalDateTime date = LocalDateTime.parse(eventResult.getValue(0, "date").toString(), formatter);
+    boolean open = Boolean.parseBoolean(eventResult.getValue(0, "open").toString());
+    int eventid = Integer.parseInt(eventResult.getValue(0, "id").toString());
+    if (eventResult.isEmpty() || !open || LocalDateTime.now().compareTo(date) >= 0)
+      return false;
 
-    if (meetingResult.isEmpty() || memberResult.isEmpty() || !open || LocalDateTime.now().compareTo(date) >= 0)
+    try {
+      db.execute("insert into event_check_ins\n" + "  (id, user_id, event_id)\n" + "  values (DEFAULT, ?, ?);", userid,
+          eventid);
+
+    } catch (Exception e) {
```

Comment:
You should never really be catching general Exceptions. Instead change this to be catching the specific exceptions you might reasonably expect to see.